### ISSUE TITLE
RIMBANET code 

### ISSUE THREAD

**Issue by bsb2014** [jnr3hh/Reed_Civelek_2023_manuscript#1]
I recently tried running your RIMBANET code posted on GitHub, and I’m pleased to report that everything worked well for the most part. However, I noticed that some isolated nodes appear in the final output file (see example below).

I wanted to check if you’ve encountered this issue before or if you have any suggestions on how to address it.

Thanks 

There is a example 
digraph G {
ABCC3->F13A1
ABCC3->TSPAN33
.........
ALK; # isolated nodes
CLEC9A; # isolated nodes
CXCL13; # isolated nodes
}

**Comment by jnr3hh** [comment]
Sorry for the late reply.

The isolated nodes are normal, the RIMBANET algorithm will only add edges
between nodes if there is enough evidence within the data to do so.
Specifically, 1000 iterations of the network are created, then the
iterations are averaged into a consensus network. Edges in the consensus
network must be present in at least 30% of the iterations. So, if a node
has few edges connecting it in the iterations, it could have no edges in
the final network. This might be because the expression of this gene
doesn't change very much, or doesn't change in a way that relates to other
genes in the network. So, it's definitely not a problem, and I usually find
that using more variable genes as the input and using a larger sample size
will result in more connected genes and fewer isolated genes in the output.
However, I also found that it is dataset dependent, so it could be that
your particular dataset results in more unconnected genes. I think this
would only be a problem if a majority of the genes were isolated nodes-
then I would consider looking into your input geneset expression and your
discretization strategy.

That being said, I usually focus only on the connected edges when
performing analyses on the networks and I only read the edges into R after
using the RIMBANET algorithm.

I hope this answers your question, let me know if I can clarify anything
else!

Best,
Jordan Reed

**Comment by bsb2014** [comment]
Hi Jordan, thanks for your explanation about the isolated nodes. 

I just wanted to double-check with you—could you confirm whether the static library libxerces-c (xerces-c-src_2_8_0) is embedded directly into your testBN program? Thanks!

**Comment by jnr3hh** [comment]
No, you have to separately download xerces, version < 2.8. I am attaching a
powerpoint that should be helpful to you, it explains the whole
installation process and running the code.

-Jordan

**Comment by bsb2014** [comment]
Hi Jordan, many thanks! However, I did not find the attachment. Could you please re-upload it :)
